nettests/nettests.go

```diff
@@ -273,6 +279,117 @@ func (c *Controller) Init(nt *mk.Nettest) error {
 	return nil
 }
 
+// Run runs the selected nettest using the related experiment
+// with the specified inputs.
+//
+// This function will continue to run in most cases but will
+// immediately halt if something's wrong with the file system.
+//
+// This is the codepath for running ooni/probe-engine nettests.
+func (c *Controller) Run(exp *experiment.Experiment, inputs []string) error {
+	ctx := context.Background()
+
+	// This will configure the controller as handler for the callbacks
+	// called by ooni/probe-engine/experiment.Experiment.
+	exp.Callbacks = handler.Callbacks(c)
+
+	c.msmts = make(map[int64]*database.Measurement)
+
+	// These values are shared by every measurement
+	reportID := sql.NullString{String: "", Valid: false}
```

Comment:
`<bikeshed>`I generally use the pattern `var reportID sql.NullString` for similar situations. It's more clear that I am initialising the variable and I am not actually setting the values to anything meaningful`</bikeshed>`

Reply:
Yes, I also use the same pattern as it's more idiomatic Go. This code has been copied from existing code in the `Init` function and didn't change it because I wanted to keep it easier for one to see how the code has changed in this refactoring process.